guides/source/getting_started.md

```diff
@@ -7,840 +7,839 @@ This guide covers getting up and running with Ruby on Rails.
+A `PUT` or `PATCH` request submits data to a URL for updating a record.
+A `DELETE` request to a URL tells the server to delete a record.
 
-### Showing a List of Articles
+### Rails Routes
 
-Let's go back to our controller in `app/controllers/articles_controller.rb`, and
-change the `index` action to fetch all articles from the database:
+A `route` in Rails refers to a line of code that matches an HTTP Method  and a URL path. The route also tells Rails which `controller` and `action` should respond to a request that matches.
+
+To define a route in Rails, we can add the following to `config/routes.rb`
 
 ```ruby
-class ArticlesController < ApplicationController
-  def index
-    @articles = Article.all
-  end
+Rails.application.routes.draw do
+  get "/products", to: "products#index"
 end
 ```
 
-Controller instance variables can be accessed by the view. That means we can
-reference `@articles` in `app/views/articles/index.html.erb`. Let's open that
-file, and replace its contents with:
+This route tells Rails to look for GET requests to the `/products` path. When Rails sees a request that matches, it will send the request to a Controller and Action for handling the request and generating a response.
 
-```html+erb
-<h1>Articles</h1>
+In this example, we specified `"products#index"` for where to route the request. This translates to a class named `ProductsController` and the `index` action inside of it. This will be responsible for handling the request and returning a response to the browser.
 
-<ul>
-  <% @articles.each do |article| %>
-    <li>
-      <%= article.title %>
-    </li>
-  <% end %>
-</ul>
+You'll notice that we don't need to specify the protocol, domain, or query params in our routes. That's basically because the protocol and domain are for making sure the request reaches your server. From there, Rails picks up the request and knows to use the path for responding to the request. The query params are like options that Rails can use to apply to the request, so they are typically used in the controller for filtering the data.
+
+Let's look at another example. Add this line after the previous route:
+
+```ruby
+post "/products", to: "products#create"
+```
+
+Here, we're telling Rails to listen to POST requests to "/products" and process those requests in the `ProductsController` with the `create` action.
+
+Routes may also need to dynamically match requests. So how does that work?
+
+```ruby
+get "/products/:id", to: "products#show"
+```
+
+This route has `:id` in it. This is called a `parameter` and it captures a portion of the URL to be used later for processing the request. If a user visits `/products/1`, the `:id` param is set to `1` and can be used in the controller action for looking up the Product record with ID of 1.
+
+Route parameters don't have to be Integers either. For example, you could have a blog with articles and match `/blog/hello-world` with the following route:
```

Comment:
Passing a note from the beginner: "This paragraph and the resulting example totally lost me. What is the "slug" command? How does it know the name of the blog you want matched."

Reply:
I know the answer is in the question here, but just wanted to pass through that this paragraph and example made her stop and question.